osutil/io_test.go

```diff
@@ -174,3 +174,61 @@ func (ts *AtomicWriteTestSuite) TestAtomicWriteFileNoOverwriteTmpExisting(c *C)
 	err = AtomicWriteFile(p, []byte(""), 0600, 0)
 	c.Assert(err, ErrorMatches, "open .*: file exists")
 }
+
+func (ts *AtomicWriteTestSuite) TestAtomicFileUidGidError(c *C) {
+	d := c.MkDir()
+	p := filepath.Join(d, "foo")
+
+	_, err := NewAtomicFile(p, 0644, 0, -1, 1)
+	c.Check(err, ErrorMatches, ".*needs none or both of uid and gid set")
+}
+
+func (ts *AtomicWriteTestSuite) TestAtomicFileChownError(c *C) {
+	if os.Getuid() == 0 {
+		c.Skip("this will fail if run as root")
+	}
+	d := c.MkDir()
+	p := filepath.Join(d, "foo")
+
+	aw, err := NewAtomicFile(p, 0644, 0, 0, 0)
+	c.Assert(err, IsNil)
+	defer aw.Cancel()
+
+	_, err = aw.Write([]byte("hello"))
+	c.Assert(err, IsNil)
+
+	c.Check(aw.Finalize(), ErrorMatches, ".* operation not permitted")
```

Comment:
This could be tester also by adding a "fdChown(fd, uid)" which might be a bit more direct than this indirect test, one upside is that we could check the arguments passed to chown (i.e. if its the right uid). But I guess I'm deep in nitpick lnad now :)